Added ability to double click to open files

sah339 opened the pull request:
Uses system 'default' app to open
Also no longer crashes if it encounters broken symlinks
 - now instead marks the result name with ' - broken symlink'
 - opening (double click) does nothing as the path for the result is set to the current directory path

sah339 committed:
Added ability to double click to open files
Uses system 'default' app tp open
Also no longer crashes if it encounters broken symlinks
 - now instead marks the result name with ' - broken symlink'
 - opening (double click) does nothing as the path for the result is set to the current directory path

JSH4a commented on backend/src/main.rs:
```diff
@@ -27,15 +27,17 @@ fn read_directory(directory_path: &str) -> String {
 
                 // Get file name
-                let file_name = entry.file_name().into_string().unwrap();
+                let mut file_name = entry.file_name().into_string().unwrap_or("Err getting dir name".to_string());
```
Just continue the loop of not there

sah339 replied:
Done

JSH4a commented on backend/src/main.rs:
```diff
@@ -27,15 +27,17 @@ fn read_directory(directory_path: &str) -> String {
-                let file_name = entry.file_name().into_string().unwrap();
+                let mut file_name = entry.file_name().into_string().unwrap_or("Err getting dir name".to_string());
+
+                let file_path = match entry.path().canonicalize() {
```
set to broken file type

JSH4a commented on backend/src/main.rs:
```diff
@@ -73,9 +75,28 @@ fn read_directory(directory_path: &str) -> String {
 
+#[tauri::command(rename_all = "snake_case")]
+fn open_file(file_path: &str) -> String {
+    match fs::canonicalize(file_path) {
```
don't need this?

sah339 replied:
removed

JSH4a commented on frontend/components/results/ResultRow.vue:
```diff
@@ -29,6 +29,10 @@ export default {
     updatePath() {
```
rename action

JSH4a commented on frontend/components/results/ResultRow.vue:
```diff
@@ -29,6 +29,10 @@ export default {
     updatePath() {
       if (this.type === "Directory") {
         this.pathEmitter.emit("path-update", this.path);
+      } else if (this.type === "File") {
```
Vue side enum to match rust one

sah339 committed:
Fixed things
Rust uses enum for file type
Added broken link type instead of crashing :)
Renamed 'UpdatePath' Vue function to be more in line with what it actually does
Removed some unnecessary stuff

JSH4a commented on backend/src/main.rs:
```diff
@@ -26,25 +35,34 @@ fn read_directory(directory_path: &str) -> String {
+                    Ok(s) => { s }
+                    Err(_) => { continue; }
+                };
 
                 // Determine file type
-                let file_type = if metadata.is_file() {
-                    "File"
-                } else if metadata.is_dir() {
-                    "Directory"
+                let mut file_type = if f_type.is_file() {
+                    FileType::File
+                } else if f_type.is_dir() {
+                    FileType::Directory
+                } else if f_type.is_symlink() {
+                    FileType::Link
                 } else {
-                    "Other"
-                }.to_string();
+                    FileType::Other
+                };
```
match

JSH4a reviewed (approved):
LGTM

sah339 committed:
Added Vue side enum (doesn't work)
